Source/WTF/wtf/RetainPtr.h

```diff
@@ -53,12 +53,6 @@
 typedef struct objc_object *id;
 #endif
 
-// Because ARC enablement is a compile-time choice, and we compile this header
-// both ways, we need a separate copy of our code when ARC is enabled.
-#if __has_feature(objc_arc)
-#define adoptNS adoptNSArc
-#endif
```

Comment:
I am surprised that it’s safe to remove this. Can you prove that it is?

Reply:
I'm relying on Tools/TestWebKitAPI/Tests/WTF/cocoa/TypeCastsCocoa.mm and Tools/TestWebKitAPI/Tests/WTF/cocoa/TypeCastsCocoaARC.mm as the test cases here.

The reason I think it's OK to have one adoptNS across ARC and non-ARC is that the function has the same declared interface and the same behavior regardless of which way you compile it. (This is not the same as, e.g., BlockPtr, which has different data members when compiled ARC vs non-ARC.)

I guess it's not that much value to remove though. Should I add it back?

Reply:
If it’s not needed, that’s fine with me. Having a test case is good. It was unpleasant before when we had it wrong!